src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -30,7 +30,9 @@ class ObjectNormalizer extends AbstractObjectNormalizer
 {
     protected $propertyAccessor;
 
-    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory = null, NameConverterInterface $nameConverter = null, PropertyAccessorInterface $propertyAccessor = null, PropertyTypeExtractorInterface $propertyTypeExtractor = null, ClassDiscriminatorResolverInterface $classDiscriminatorResolver = null)
+    private static $isDiscriminatorAttributeCache = array();
```

Comment:
I would not make this property static, because injected resolver can be different form an instance to another.